src/governance/agent.js

```diff
@@ -179,40 +188,105 @@ function proposalThresholdSetFinding(address, config, oldThresh, newThresh) {
+  }
+  if (!isFilledString(config.protocolAbbreviation)) {
+      errMsg = `protocolAbbreviation required`;
+      return { ok, errMsg };
+  }
+
+  for (const [name, entry] of Object.entries(config.contracts)) {
+    if (!isObject(entry) || isEmptyObject(entry)) {
+      errMsg = `governance key required`;
+      return { ok, errMsg };
+    }
+
+    const { abiFile } = entry.governance;
+    const { address } = entry;
+
+    if (address === undefined) {
+      errMsg = `No address found in configuration file for '${name}'`;
+      return { ok, errMsg };
+    }
+
+    if (abiFile === undefined) {
+      errMsg = `No ABI file found in configuration file for '${name}'`;
+      return { ok, errMsg };
+    }
+
+    // check that the address is a valid address
+    if (!ethers.utils.isHexString(address, 20)) {
+      errMsg = `invalid address`;
+      return { ok, errMsg };
+    }
+
+    // load the ABI from the specified file
+    // the call to getAbi will fail if the file does not exist
+    const abi = getInternalAbi(config.agentType, abiFile);
+
+    // extract all of the event names from the ABI
+    const events = getObjectsFromAbi(abi, 'event');
+
+    // verify that at least the minimum list of supported events are present
+    for (let i = 0; i < MINIMUM_EVENT_LIST.length; i++) {
+      const eventName = MINIMUM_EVENT_LIST[i];
+      if (Object.keys(events).indexOf(eventName) === -1) {
+        errMsg = `ABI does not contain minimum supported event: ${eventName}`;
+        return { ok, errMsg };
+      }
+    }
+  }
+
+  ok = true;
+  return {ok, errMsg};
+};
+
 const initialize = async (config) => {
-	let agentState = {};
-
-	agentState.config = {
-		developerAbbreviation: config.developerAbbreviation,
-		protocolName: config.protocolName,
-		protocolAbbreviation: config.protocolAbbreviation,
-	};
-	agentState.goverance = config.contracts;
-	agentState.contracts = Object.entries(agentState.goverance).map(([name, entry]) => {
-		const { governance: { abiFile }, address } = entry;
-
-		if (address === undefined) {
-			throw new Error(`No address found in configuration file for '${name}'`);
-		}
-
-		if (abiFile === undefined) {
-			throw new Error(`No ABI file found in configuration file for '${name}'`);
-		}
-
-		const abi = getInternalAbi(config.agentType, abiFile);
-		const iface = new ethers.utils.Interface(abi);
-		const names = Object.keys(iface.events);
-		const ftype = ethers.utils.FormatTypes.full;
-		const eventSignatures = names.map((eventName) => iface.getEvent(eventName).format(ftype));
-
-		const contract = {
-			address,
-			eventSignatures,
-		};
-		return contract;
-	});
-
-	return agentState;
+  let agentState = {...config};
+
+  const { ok, errMsg } = validateConfig(config);
+  if (!ok) {
+    throw new Error(errMsg);
+  }
+
+  agentState.developerAbbreviation = config.developerAbbreviation;
+  agentState.protocolName = config.protocolName;
+  agentState.protocolAbbreviation = config.protocolAbbreviation;
```

Comment:
These lines can be removed because they are redundant (line 262 already copies the key/value pairs from the `config` Object to the `agentState` Object).
```suggestion
```